File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        zchange = ochange = 0
        zp = '0'
        op = '1'
        for i in range(len(s)):
            if(zp != s[i]):
                zchange += 1
            else:
                ochange += 1
            
            zp, op = op, zp
        
        return min(zchange, ochange)
        
            
            
            

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.go
func minOperations(s string) int {
    return min(findRes(s, '1'), findRes(s, '0'))
}

func findRes(s string, ch rune) int {
    res := 0
    for _, c := range s {
        if c != ch {
            res++
        }
        if ch == '0' {
            ch = '1'
        } else { ch = '0' }
    }
    return res
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}
